repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java

```diff
@@ -2403,6 +2404,77 @@ private void recordOutGoingEdgesToBeDeleted(Iterable<AtlasEdge> outgoingEdges) t
                 AtlasRelationshipStoreV2.recordRelationshipMutation(AtlasRelationshipStoreV2.RelationshipMutation.RELATIONSHIP_HARD_DELETE, edge, entityRetriever);
         }
     }
+
+    @Override
+    @GraphTransaction
+    public void repairMeaningAttributeForTerms(List<String> termGuid) {
+
+        for (String guid : termGuid) {
+            LOG.info(" term guid " + guid);
+
+            AtlasVertex termVertex = AtlasGraphUtilsV2.findByGuid(this.graph, guid);
+
+            if(termVertex!= null && ATLAS_GLOSSARY_TERM_ENTITY_TYPE.equals(getTypeName(termVertex)) &&
+                    GraphHelper.getStatus(termVertex) == AtlasEntity.Status.ACTIVE) {
+                Iterable<AtlasEdge> edges = termVertex.getEdges(AtlasEdgeDirection.OUT, Constants.TERM_ASSIGNMENT_LABEL);
+                // Get entity to tagged with term.
+                if (edges != null) {
+                    for (Iterator<AtlasEdge> iter = edges.iterator(); iter.hasNext(); ) {
+                        AtlasEdge edge = iter.next();
+                        if (GraphHelper.getStatus(edge) == AtlasEntity.Status.ACTIVE) {
+                            AtlasVertex entityVertex = edge.getInVertex();
+                            if (entityVertex != null & getStatus(entityVertex) == AtlasEntity.Status.ACTIVE) {
+                                repairMeanings(entityVertex);
```

Comment:
In case we are going to run this API for multiple terms, it is clear that you will run API with a list all term GUID
Given this, there can be a scenario where an entityVertex can appear for different terms here, updating same entity again will not needed as we are already iterating all terms & updating entityVertex

Can we maintain a list here of processed entityVertices (across the terms) & skip in case it is already processed

Reply:
Added fix to review comment.